client/allocrunner/group_service_hook.go

```diff
@@ -267,7 +268,9 @@ func (h *groupServiceHook) getWorkloadServicesLocked() *serviceregistration.Work
 
 	tokens := map[string]string{}
 	for _, service := range h.services {
-		if token, ok := allocTokens[service.Cluster][service.MakeUniqueIdentityName()]; ok {
+		id := fmt.Sprintf("%s_%s",
+			structs.ConsulServiceIdentityNamePrefix, service.MakeUniqueIdentityName())
+		if token, ok := allocTokens[service.Cluster][id]; ok {
```

Comment:
I feel like we have this logic sprinkled in several places. Are there scenarios where we would want to use the result of `MakeUniqueIdentityName` without the identity prefix? Otherwise we could have it so `MakeUniqueIdentityName` already returns the prefixed identity name.

Reply:
I think you're right... I just did a quick lookup and all callers of `(*Service) MakeUniqueIdentityName` are formatting it with the prefix. I can update the method and all callers.

Reply:
Done in 662f80b